src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlDataSource.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NET8_0_OR_GREATER 
+
+using Microsoft.Data.SqlClient;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Data.SqlClientX
+{
+    internal abstract class SqlDataSource : DbDataSource
```

Comment:
Should this be a public class?

Reply:
As discussed, we'll keep this internal for now until we're ready to support a broader public API surface.